### DISCUSSION TITLE
Support for pysqlite3

### DISCUSSION THREAD

**Discussion by AbdealiLoKo** [sqlalchemy/sqlalchemy#8440] [Ideas]
Hi, I was recently figuring out how to use the latest sqlite in my application to avoid differences in the OS's sqlite (Red Hat 7 has a very old sqlite)
And I came across: https://pypi.org/project/pysqlite3-binary/
Which was a super easy 1 step install to get the latest sqlite3 in my flask application.

But when I tried using it with sqlalchemy it was giving me an error i guess because of:
https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_4_40/lib/sqlalchemy/dialects/sqlite/pysqlite.py#L473
Where the import is just doing: `from sqlite3 import dbapi2 as sqlite`
And the library needs: `from pysqlite3 import dbapi2 as sqlite`

I was wondering if there would be any interest to create a new `sqlite+pysqlite3://` option to allow using this ?
(Similar to the `mysql+pymysql://` ?)

I did find: https://github.com/hoehrmann/sqlalchemy-pysqlite3
But was wondering if we could add it to sqlalchemy itself - cause it seems quite simpler

**Comment by CaselIT** [comment]
Hi,

if that package has the same api as the builtin module you can just use the [`module`](https://docs.sqlalchemy.org/en/14/core/engines.html?highlight=create_engine#sqlalchemy.create_engine.params.module) argument of create engine

**Reply by zzzeek** [reply]
agree, it's the same dialect, I think `module` is best

**Reply by AbdealiLoKo** [reply]
Perfect. Wasn't aware of this option - Thanks for the swift response !

**Reply by AbdealiLoKo** [reply]
Hi, I was revisiting some code, and I was wondering if this is still the best option.
Specifically I see some code like [_fix_sqlite_issue_99953](https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_45/lib/sqlalchemy/dialects/sqlite/pysqlite.py#L666) which is doing a `import sqlite3` directly and not using the "module" that I provide (it is noted as a test class though...)

I haven't seen any issues ... but just wondering whether `module` is still the best way of doing this in sqlalchemy 2.x

**Reply by CaselIT** [reply]
Thar function is in a class named _SQLiteDialect_pysqlite_numeric that is documented as

>  internal use only.  This dialect is **NOT** supported by SQLAlchemy
    and may change at any time.

So you are not supposed to use it
